client.go

```diff
@@ -1353,3 +1401,555 @@ func (c *ToznySDKV3) UnbrokerShare(ctx context.Context, authorizerClientID strin
+		}
+		// make the capabilities
+		groupMemberCapabilities := []string{storageClient.ShareContentGroupCapability, storageClient.ReadContentGroupCapability}
+		// make a new member
+		memberReq := []storageClient.GroupMember{}
+		memberReq = append(memberReq, storageClient.GroupMember{
+			ClientID:        uuid.MustParse(c.StorageClient.ClientID),
+			MembershipKey:   membershipKeyResp,
+			CapabilityNames: groupMemberCapabilities,
+		})
+		// add group members
+		addMemberReq := storageClient.AddGroupMembersRequest{
+			GroupID:      createGroupResponse.GroupID,
+			GroupMembers: memberReq,
+		}
+		_, err = c.StorageClient.AddGroupMembers(ctx, addMemberReq)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// the group already exists
+		group = &responseList.Groups[0]
+	}
+	return group, nil
+}
+
+// ValidateSecret checks that the secret contains valid input for each entry
+func ValidateSecret(secret CreateSecretOptions) error {
+	secret.SecretName = strings.TrimSpace(secret.SecretName)
+	secret.SecretValue = strings.TrimSpace(secret.SecretValue)
+	if secret.SecretType == "" {
+		return errors.New("type cannot be empty")
+	}
+	if !SliceContainsString(SECRET_TYPES, secret.SecretType) {
+		return errors.New("invalid type")
+	}
+	if secret.SecretName == "" {
+		return errors.New("name cannot be empty")
+	}
+	matched, err := regexp.MatchString(`^[a-zA-Z0-9-_]{1,50}$`, secret.SecretName)
+	if err != nil {
+		return err
+	}
+	if !matched {
+		return errors.New("Secret name must contain 1-50 alphanumeric characters, -, or _")
+	}
+	if secret.SecretValue == "" && secret.SecretType == "Credential" {
+		return errors.New("Value cannot be empty")
+	}
+	if secret.SecretType == "File" && strings.TrimSpace(secret.SecretName) == "" {
+		return errors.New("File name cannot be empty")
+	}
+	if secret.SecretType == "Client" {
+		err := VerifyRawClientCredentials(secret.SecretValue)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// EncryptAccessKeyForGroup encrypts the access key and returns the result
+func EncryptAccessKeyForGroup(member storageClient.StorageClient, ak e3dbClients.SymmetricKey) (string, error) {
+	encryptionKeys := e3dbClients.EncryptionKeys{
+		Public: e3dbClients.Key{
+			Type:     e3dbClients.DefaultEncryptionKeyType,
+			Material: member.EncryptionKeys.Public.Material,
+		},
+		Private: member.EncryptionKeys.Private,
+	}
+	eak, eakN, err := e3dbClients.BoxEncryptToBase64(ak[:], encryptionKeys)
+	if err != nil {
+		return "", err
+	}
+	wrappedAccessKey := fmt.Sprintf("%s.%s", eak, eakN)
+	return wrappedAccessKey, nil
+}
+
+// SliceContainsString checks if str is present in an array of strings
+func SliceContainsString(list []string, str string) bool {
+	for _, v := range list {
+		if v == str {
+			return true
+		}
+	}
+	return false
+}
+
+// VerifyRawClientCredentials ensures that client credentials are in the proper form and contain all required keys
+func VerifyRawClientCredentials(credential string) error {
```

Comment:
This method should take the input and use Go's builtin json parsing tools to create a client struct
```go
var clientConfig ClientConfig
err := json.Unmarshall(credential, &ClientConfig)
if err ...
```
Additionally for validating the key lengths, we should be reencoding them into byte-slices and checking those lengths
for that you can use 

```go
byteslice, err := Base64Decode(base64EncodedKey)
if err...
len(pskBytes) == expectedKeyLength
```